finetune/util/huggingface_interface.py

```diff
@@ -88,16 +90,31 @@ def finetune_featurizer(X, encoder, config, train=False, reuse=None, **kwargs):
             call_args = tf_inspect.getargspec(hf_model).args
             kwargs = {k: v for k, v in kwargs.items() if k in call_args}
 
+            if hf_config_instance.is_encoder_decoder:
+                # Not the right decision universally -- more broadly may
+                # condition on some secondary text to nudge the model into producing
+                # the desired output
+                kwargs["decoder_input_ids"] = X
```

Comment:
I would be inclined to have T5 locked into a single downstream model similar to the old Seq2Seq model type. I think trying to make it fit each of the model types will be overly complicated.